test_move_cut fails on Python 3.14

FYI:

```py
=================================== FAILURES ===================================
________________________________ test_move_cut _________________________________

    def test_move_cut():
        da = scatter()
        fig = scatter3dfigure(Node(da), x='x', y='y', z='z', cbar=True)
        clip = ClippingPlanes(fig)
        clip.add_x_cut.click()
        xcut = clip.cuts[-1]
        assert xcut.outlines[0].position[0] == xcut.slider.value[0]
        assert xcut.outlines[1].position[0] == xcut.slider.value[1]
        pts = list(fig.artists.values())[-1]
        npoints = pts._data.shape[0]
>       xcut.slider.value = [xcut.slider.min, xcut.slider.value[1]]
        ^^^^^^^^^^^^^^^^^

tests/widgets/clip3d_test.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/i7lysg4h0dccnxkxx2m2pbn67pdzlzm4-python3.14-traitlets-5.14.3/lib/python3.14/site-packages/traitlets/traitlets.py:716: in __set__
    self.set(obj, value)
/nix/store/i7lysg4h0dccnxkxx2m2pbn67pdzlzm4-python3.14-traitlets-5.14.3/lib/python3.14/site-packages/traitlets/traitlets.py:706: in set
    obj._notify_trait(self.name, old_value, new_value)
/nix/store/i7lysg4h0dccnxkxx2m2pbn67pdzlzm4-python3.14-traitlets-5.14.3/lib/python3.14/site-packages/traitlets/traitlets.py:1513: in _notify_trait
    self.notify_change(
/nix/store/ks44k0nb3ka3c92qhav1w4a5w1hfvkcd-python3.14-ipywidgets-8.1.7/lib/python3.14/site-packages/ipywidgets/widgets/widget.py:701: in notify_change
    super().notify_change(change)
/nix/store/i7lysg4h0dccnxkxx2m2pbn67pdzlzm4-python3.14-traitlets-5.14.3/lib/python3.14/site-packages/traitlets/traitlets.py:1525: in notify_change
    return self._notify_observers(change)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/nix/store/i7lysg4h0dccnxkxx2m2pbn67pdzlzm4-python3.14-traitlets-5.14.3/lib/python3.14/site-packages/traitlets/traitlets.py:1568: in _notify_observers
    c(event)
/nix/store/dqbgkmik163hrjsdcwa4qf3l28yw1lws-python3.14-plopp-25.11.0/lib/python3.14/site-packages/plopp/widgets/clip3d.py:177: in move
    self._throttled_update()
/nix/store/dqbgkmik163hrjsdcwa4qf3l28yw1lws-python3.14-plopp-25.11.0/lib/python3.14/site-packages/plopp/widgets/debounce.py:51: in debounced
    timer.start()
/nix/store/dqbgkmik163hrjsdcwa4qf3l28yw1lws-python3.14-plopp-25.11.0/lib/python3.14/site-packages/plopp/widgets/debounce.py:23: in start
    self._task = asyncio.ensure_future(self._job())
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/lib/python3.14/asyncio/tasks.py:730: in ensure_future
    loop = events.get_event_loop()
           ^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <asyncio.unix_events._UnixDefaultEventLoopPolicy object at 0x7ffdf6f78ad0>

    def get_event_loop(self):
        """Get the event loop for the current context.
    
        Returns an instance of EventLoop or raises an exception.
        """
        if self._local._loop is None:
>           raise RuntimeError('There is no current event loop in thread %r.'
                               % threading.current_thread().name)
E           RuntimeError: There is no current event loop in thread 'MainThread'.

/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/lib/python3.14/asyncio/events.py:715: RuntimeError
=========================== short test summary info ============================
SKIPPED [5] tests/plotting/plot_1d_test.py:24: Skipping because kaleido is not installed
SKIPPED [1] tests/plotting/plot_1d_test.py:360: Log scale with datetime not supported in plotly
SKIPPED [1] tests/plotting/plot_1d_test.py:375: Log scale with datetime not supported in plotly
SKIPPED [1] tests/plotting/plot_1d_test.py:390: Log scale with datetime not supported in plotly
FAILED tests/widgets/clip3d_test.py::test_move_cut - RuntimeError: There is no current event loop in thread 'MainThread'.
================== 1 failed, 1204 passed, 8 skipped in 54.87s ==================
Traceback (most recent call last):
  File "/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/lib/python3.14/_py_warnings.py", line 856, in _warn_unawaited_coroutine
    _wm.warn(
    ~~~~~~~~^
        msg, category=RuntimeWarning, stacklevel=2, source=coro
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
RuntimeWarning: coroutine 'Timer._job' was never awaited

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/nix/store/87niicfndnfqsi1wvjkzzh05fg8y7jhb-python3.14-pytest-8.4.2/lib/python3.14/site-packages/pytest/__main__.py", line 9, in <module>
    raise SystemExit(pytest.console_main())
                     ~~~~~~~~~~~~~~~~~~~^^
  File "/nix/store/87niicfndnfqsi1wvjkzzh05fg8y7jhb-python3.14-pytest-8.4.2/lib/python3.14/site-packages/_pytest/config/__init__.py", line 201, in console_main
    code = main()
  File "/nix/store/87niicfndnfqsi1wvjkzzh05fg8y7jhb-python3.14-pytest-8.4.2/lib/python3.14/site-packages/_pytest/config/__init__.py", line 175, in main
    ret: ExitCode | int = config.hook.pytest_cmdline_main(config=config)
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/nix/store/96vzxj35a0jjk3cwrp9jd90agvs3zmmh-python3.14-pluggy-1.6.0/lib/python3.14/site-packages/pluggy/_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/96vzxj35a0jjk3cwrp9jd90agvs3zmmh-python3.14-pluggy-1.6.0/lib/python3.14/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/96vzxj35a0jjk3cwrp9jd90agvs3zmmh-python3.14-pluggy-1.6.0/lib/python3.14/site-packages/pluggy/_callers.py", line 167, in _multicall
    raise exception
  File "/nix/store/96vzxj35a0jjk3cwrp9jd90agvs3zmmh-python3.14-pluggy-1.6.0/lib/python3.14/site-packages/pluggy/_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "/nix/store/87niicfndnfqsi1wvjkzzh05fg8y7jhb-python3.14-pytest-8.4.2/lib/python3.14/site-packages/_pytest/main.py", line 336, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/nix/store/87niicfndnfqsi1wvjkzzh05fg8y7jhb-python3.14-pytest-8.4.2/lib/python3.14/site-packages/_pytest/main.py", line 331, in wrap_session
    config._ensure_unconfigure()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/nix/store/87niicfndnfqsi1wvjkzzh05fg8y7jhb-python3.14-pytest-8.4.2/lib/python3.14/site-packages/_pytest/config/__init__.py", line 1131, in _ensure_unconfigure
    self._cleanup_stack.close()
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/lib/python3.14/contextlib.py", line 627, in close
    self.__exit__(None, None, None)
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/lib/python3.14/contextlib.py", line 619, in __exit__
    raise exc
  File "/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/lib/python3.14/contextlib.py", line 604, in __exit__
    if cb(*exc_details):
       ~~^^^^^^^^^^^^^^
  File "/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/lib/python3.14/contextlib.py", line 482, in _exit_wrapper
    callback(*args, **kwds)
    ~~~~~~~~^^^^^^^^^^^^^^^
  File "/nix/store/87niicfndnfqsi1wvjkzzh05fg8y7jhb-python3.14-pytest-8.4.2/lib/python3.14/site-packages/_pytest/unraisableexception.py", line 95, in cleanup
    collect_unraisable(config)
    ~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/nix/store/87niicfndnfqsi1wvjkzzh05fg8y7jhb-python3.14-pytest-8.4.2/lib/python3.14/site-packages/_pytest/unraisableexception.py", line 79, in collect_unraisable
    raise errors[0]
  File "/nix/store/87niicfndnfqsi1wvjkzzh05fg8y7jhb-python3.14-pytest-8.4.2/lib/python3.14/site-packages/_pytest/unraisableexception.py", line 67, in collect_unraisable
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pytest.PytestUnraisableExceptionWarning: Exception ignored while finalizing coroutine <coroutine object Timer._job at 0x7ffdfe6f8ee0>: None
```

Reference:

-  https://github.com/NixOS/nixpkgs/pull/482111

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -80,7 +80,6 @@ filterwarnings = [
   'ignore:Passing unrecognized arguments to super:DeprecationWarning',
   'ignore:Jupyter is migrating its paths:DeprecationWarning',
   'ignore:setDaemon\(\) is deprecated, set the daemon attribute instead:DeprecationWarning',
-  'ignore:There is no current event loop:DeprecationWarning',
   # Should be removed once https://github.com/ipython/ipykernel/pull/1248 is released
   'ignore:Parsing dates involving a day of month without a year:DeprecationWarning',
   # Should be removed once we lower pin a matplotlib version that respects the deprecation

--- a/src/plopp/widgets/clip3d.py
+++ b/src/plopp/widgets/clip3d.py
@@ -74,7 +74,7 @@ class Clip3dTool(ipw.HBox):
         self._unit = self._limits[axis].unit
         self.visible = True
         self._update = update
-        self._throttled_update = debounce(update, wait=0.3)
+        self._throttled_update = debounce(update, wait=0.2)
         self._border_visible = border_visible
 
         w_axis = 2 if self.kind == 'x' else 0

--- a/src/plopp/widgets/clip3d.py
+++ b/src/plopp/widgets/clip3d.py
@@ -212,7 +212,7 @@ class ClipValueTool(ipw.HBox):
         self._unit = self._limits.unit
         self.visible = True
         self._update = update
-        self._throttled_update = debounce(update, wait=0.3)
+        self._throttled_update = debounce(update, wait=0.2)
         self.kind = 'v'
 
         center = self._limits.mean().value

--- a/src/plopp/widgets/clip3d.py
+++ b/src/plopp/widgets/clip3d.py
@@ -74,6 +74,7 @@ class Clip3dTool(ipw.HBox):
         self._unit = self._limits[axis].unit
         self.visible = True
         self._update = update
+        self._throttled_update = debounce(update, wait=0.3)
         self._border_visible = border_visible
 
         w_axis = 2 if self.kind == 'x' else 0
@@ -189,10 +190,6 @@ class Clip3dTool(ipw.HBox):
         xmin, xmax = self.range
         return (da.coords[self.dim] >= xmin) & (da.coords[self.dim] < xmax)
 
-    @debounce(0.3)
-    def _throttled_update(self):
-        self._update()
-
 
 class ClipValueTool(ipw.HBox):
     """
@@ -215,6 +212,7 @@ class ClipValueTool(ipw.HBox):
         self._unit = self._limits.unit
         self.visible = True
         self._update = update
+        self._throttled_update = debounce(update, wait=0.3)
         self.kind = 'v'
 
         center = self._limits.mean().value
@@ -279,10 +277,6 @@ class ClipValueTool(ipw.HBox):
         ):
             return
         self._throttled_update()
-
-    @debounce(0.3)
-    def _throttled_update(self):
-        self._update()
 
 
 class ClippingManager(ipw.HBox):

--- a/src/plopp/widgets/debounce.py
+++ b/src/plopp/widgets/debounce.py
@@ -3,53 +3,36 @@
 
 import asyncio
 from collections.abc import Callable
+from functools import wraps
 
 
-class Timer:
+def debounce(fn: Callable, *, wait: float):
     """
-    From:
-    https://ipywidgets.readthedocs.io/en/8.0.2/examples/Widget%20Events.html#Debouncing
+    Wrap a function so that its execution is postponed until `wait` seconds have
+    elapsed since the last time it was invoked.
+
+    If there is no running event loop, the function is called immediately. This is
+    useful in synchronous contexts such as tests and scripts, where delayed execution
+    cannot be scheduled without blocking or using another thread.
     """
+    handle: asyncio.TimerHandle | None = None
 
-    def __init__(self, timeout: float, callback: Callable):
-        self._timeout = timeout
-        self._callback = callback
+    @wraps(fn)
+    def debounced(*args, **kwargs):
+        nonlocal handle
 
-    async def _job(self):
-        await asyncio.sleep(self._timeout)
-        self._callback()
+        def call_it():
+            nonlocal handle
+            handle = None
+            fn(*args, **kwargs)
 
-    def start(self):
-        self._task = asyncio.ensure_future(self._job())
+        if handle is not None:
+            handle.cancel()
+        try:
+            loop = asyncio.get_running_loop()
+        except RuntimeError:
+            call_it()
+        else:
+            handle = loop.call_later(wait, call_it)
 
-    def cancel(self):
-        self._task.cancel()
-
-
-def debounce(wait: float):
-    """
-    Decorator that will postpone a function's
-    execution until after `wait` seconds
-    have elapsed since the last time it was invoked.
-
-    From:
-    https://ipywidgets.readthedocs.io/en/8.0.2/examples/Widget%20Events.html#Debouncing
-    """
-
-    def decorator(fn: Callable):
-        timer = None
-
-        def debounced(*args, **kwargs):
-            nonlocal timer
-
-            def call_it():
-                fn(*args, **kwargs)
-
-            if timer is not None:
-                timer.cancel()
-            timer = Timer(wait, call_it)
-            timer.start()
-
-        return debounced
-
-    return decorator
+    return debounced

--- a/tests/widgets/clip3d_test.py
+++ b/tests/widgets/clip3d_test.py
@@ -60,7 +60,6 @@ def test_value_cuts(multiple_nodes):
     vcut = clip.cuts[-1]
     npoints = list(fig.artists.values())[-1]._data.shape[0]
     vcut.slider.value = [vcut.slider.min, vcut.slider.value[1]]
-    clip.update_state()  # Need to manually update state due to debounce mechanism
     # We should now have more points in the cut than before because the range is wider
     npoints2 = list(fig.artists.values())[-1]._data.shape[0]
     assert npoints2 > npoints
@@ -73,7 +72,6 @@ def test_value_cuts(multiple_nodes):
         0.5 * (vcut2.slider.value[1] + vcut2.slider.max),
         vcut2.slider.max,
     ]
-    clip.update_state()  # Need to manually update state due to debounce mechanism
     # We should now have more points in the cut than before because the range is wider
     npoints3 = list(fig.artists.values())[-1]._data.shape[0]
     assert npoints3 > npoints2
@@ -130,7 +128,6 @@ def test_move_cut():
     xcut.slider.value = [xcut.slider.min, xcut.slider.value[1]]
     assert xcut.outlines[0].position[0] == xcut.slider.value[0]
     assert xcut.outlines[1].position[0] == xcut.slider.value[1]
-    clip.update_state()  # Need to manually update state due to debounce mechanism
     new_pts = list(fig.artists.values())[-1]
     assert npoints < new_pts._data.shape[0]
 

--- a/tests/widgets/debounce_test.py
+++ b/tests/widgets/debounce_test.py
@@ -1,0 +1,58 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2026 Scipp contributors (https://github.com/scipp)
+
+import asyncio
+
+from plopp.widgets.debounce import debounce
+
+
+def test_calls_immediately_without_running_event_loop():
+    calls = []
+
+    def callback(value):
+        calls.append(value)
+
+    debounced = debounce(callback, wait=0.01)
+    debounced('first')
+    debounced('second')
+
+    assert calls == ['first', 'second']
+
+
+def test_delays_call_and_keeps_latest_value_with_running_event_loop():
+    async def run():
+        calls = []
+        called = asyncio.Event()
+
+        def callback(value):
+            calls.append(value)
+            called.set()
+
+        debounced = debounce(callback, wait=0.01)
+        debounced('discarded')
+        debounced('called')
+        assert calls == []
+        await asyncio.wait_for(called.wait(), timeout=1.0)
+        return calls
+
+    assert asyncio.run(run()) == ['called']
+
+
+def test_independent_callbacks_do_not_cancel_each_other():
+    async def run():
+        calls = []
+        both_called = asyncio.Event()
+
+        def callback(value):
+            calls.append(value)
+            if len(calls) == 2:
+                both_called.set()
+
+        first = debounce(callback, wait=0.01)
+        second = debounce(callback, wait=0.01)
+        first('first')
+        second('second')
+        await asyncio.wait_for(both_called.wait(), timeout=1.0)
+        return calls
+
+    assert sorted(asyncio.run(run())) == ['first', 'second']

